File: InterconnectBackend/Models/Requests/ConnectEntitiesRequest.cs
using Models.Enums;

namespace Models.Requests
{
    public class ConnectEntitiesRequest
    {
        public required int SourceEntityId { get; set; }
        public EntityType SourceEntityType { get; set; }
        public required int DestinationEntityId { get; set; }
        public EntityType DestinationEntityType { get; set; }
    }
}

File: InterconnectFrontend/src/api/requests/ConnectEntitiesRequest.ts
import BaseRequest from "./BaseRequest.ts";
import { EntityType } from "../../models/enums/EntityType.ts";

/**
 * Request object for creating a connection between two entities.
 */
interface ConnectEntitiesRequest extends BaseRequest {
    /** ID of the source entity */
    sourceEntityId: number;
    /** Type of the source entity */
    sourceEntityType: EntityType;
    /** ID of the destination entity */
    destinationEntityId: number;
    /** Type of the destination entity */
    destinationEntityType: EntityType;
}

export default ConnectEntitiesRequest;
